### ISSUE TITLE
Animated Image does not play completely in SimpleDraweeSpanTextView

### ISSUE THREAD

**Issue by angarron** [facebook/fresco#1940]
### Description

`SimpleDraweeSpanTextView`, when given an animated image as the source, only plays a couple frames and then freezes. Rerenders can trigger a different frame to be displayed, but it is still not actually "playing". The same happens on both `.webm` and `.gif` files.

GIF example showcasing frozen GIF using keyboard show/hide toggle to force rerenders that show new frames:

![output](https://user-images.githubusercontent.com/8857134/31738618-1e13762e-b400-11e7-97ee-05e095c8d322.gif)

### Reproduction

Try to put a GIF into a `DraweeSpanStringBuilder`.

Sample project: https://github.com/angarron/FrescoSamples/blob/master/app/src/main/java/com/agarron/frescosamples/MainActivity.kt

Sample code snippet:

```
  private fun renderImage() {
    val builder = DraweeSpanStringBuilder("Animated Image: ")

    val startIndex = builder.length
    builder.append("<image>")

    val controller = Fresco.newDraweeControllerBuilder()
        .setUri("http://cultofthepartyparrot.com/parrots/hd/parrot.gif")
        .setAutoPlayAnimations(true)
        .build()

    val draweeHierarchy = GenericDraweeHierarchyBuilder.newInstance(resources)
        .setPlaceholderImage(ColorDrawable(Color.TRANSPARENT))
        .setActualImageScaleType(ScalingUtils.ScaleType.FIT_CENTER)
        .build()

    builder.setImageSpan(
        this,
        draweeHierarchy,
        controller,
        startIndex,
        builder.length - 1,
        dpToPixels(IMAGE_WIDTH_DP),
        dpToPixels(IMAGE_HEIGHT_DP),
        false,
        DraweeSpan.ALIGN_CENTER)

    simpleDraweeSpanTextView.setDraweeSpanStringBuilder(builder)
  }
```

### Solution

I don't know a solution here yet :( Would like to dedicate the resources soon but happy to accept any tips or help in the meantime that may make that easier.

### Additional Information

* Fresco version: 1.5.0

**Comment by oprisnik** [comment]
Thanks for the bug report! This is weird. The Drawable.Callback, which is used to trigger invalidations for example when the image has been downloaded / decoded should also work for animated images. 

One problem that I had when implementing DraweeSpans was that without a custom Drawable.Callback, the wrong region might be invalidated. But this should not be the case here.

**Comment by mrkcsc** [comment]
Hi - I was just wondering if the recent commit https://github.com/facebook/fresco/commit/03a14ab06a9282ce098420217a9ca27a6ae6d587 to adjust animation listeners for infinitely animated `webp` in any way addresses this issue.

Thanks!

**Comment by oprisnik** [comment]
I think it shouldn't since these listeners are not used for invalidating the webp (and normal animated WebPs worked). I didn't have time to look into this yet, but debugging the issue shouldn't be too hard in case somebody wants to dig in and submit a pull request. There is a sample in the Showcase app.

**Comment by angarron** [comment]
This seems to be addressed in https://github.com/facebook/fresco/pull/1966 -- is there an upcoming release planned so that we can verify? I'm working on building Fresco locally and using it in my project, but if a release is on the way out I can hold off on that. Thanks so much @Kazemir and @oprisnik !

**Comment by Kazemir** [comment]
If you want to use this fix in your project, then just copy [this class](https://raw.githubusercontent.com/facebook/fresco/e548e50967768fbfdf27d6c2c8f3601533f5e11b/drawee-span/src/main/java/com/facebook/drawee/span/DraweeSpanStringBuilder.java) in your project. You should rename it and extend from DraweeSpanStringBuilder. Then use it insteed of DraweeSpanStringBuilder.

**Comment by oprisnik** [comment]
As mentioned, this was fixed by @Kazemir  in #1966.
